python/taichi/lang/type_factory_impl.py

```diff
@@ -0,0 +1,32 @@
+from taichi.core import util as cutil
```

Comment:
We have `type_factory` both as a global variable and a module name, so renaming this. Same for `quant.py` --> `quant_impl.py`